File: frontend/src/types/post.ts
export type PostStyle = 'casual' | 'breaking' | 'analysis';

export interface GeneratedPost {
  id: string;
  style: PostStyle;
  styleLabel: string;
  styleIcon: string;
  content: string;
  topicId: string;
}

export type PostStatus = 'draft' | 'scheduled' | 'published' | 'failed';

export interface ScheduledPost {
  id: string;
  content: string;
  topicId: string | null;
  topicName: string | null;
  status: PostStatus;
  scheduledAt: string | null;
  publishedAt: string | null;
  failedAt: string | null;
  errorMessage: string | null;
  createdAt: string;
  updatedAt: string;
  characterCount: number;
}

export interface PostHistory {
  id: string;
  content: string;
  topicId: string | null;
  topicName: string | null;
  publishedAt: string;
  likes: number;
  retweets: number;
  replies: number;
  views: number;
  tweetUrl: string | null;
}

export interface PostStats {
  totalPublished: number;
  totalScheduled: number;
  totalDrafts: number;
  totalFailed: number;
  thisMonthPublished: number;
}

File: backend/internal/domain/entity/post.go
package entity

import "time"

// PostStatus represents the lifecycle state of a post.
type PostStatus int

const (
	PostDraft     PostStatus = 1
	PostScheduled PostStatus = 2
	PostPublished PostStatus = 3
	PostFailed    PostStatus = 4
)

// Post represents a user's post (draft, scheduled, published, or failed).
type Post struct {
	ID           string
	UserID       string
	Content      string
	TopicID      *string
	TopicName    *string
	Status       PostStatus
	ScheduledAt  *time.Time
	PublishedAt  *time.Time
	FailedAt     *time.Time
	ErrorMessage *string
	TweetURL     *string
	Likes        int32
	Retweets     int32
	Replies      int32
	Views        int32
	CreatedAt    time.Time
	UpdatedAt    time.Time
}
